### PR TITLE
Add support for Snowflake database

### PR DESCRIPTION
**Pull request by mbobrovsky**

### CODE REVIEW

File: sql_runner/snowflake_target.go [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+		if len(strings.TrimSpace(script)) > 0 {
+
+			res, err = sft.Client.Exec(script)
+
+			if err == nil {
+				aff, _ := res.RowsAffected()
+				affected += aff
+			}
+		}
+	}
```
Comment (BenFradet) [inline sql_runner/snowflake_target.go]: problem here is that we only return the last error, what we can do to remedy that is either to:

- not split on `;` if that's possible and execute only one query with all the different statements
- accumulate the errors
- short-circuit on the first error

File: sql_runner/snowflake_target.go [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+	return QueryStatus{query, query.Path, int(affected), err}
```
Comment (BenFradet) [inline sql_runner/snowflake_target.go]: should we rather change the type of affected in `QueryStatus` to `int64`?

File: sql_runner/snowflake_target.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+		Database:     target.Database,
+		Warehouse:    target.Warehouse,
+		LoginTimeout: loginTimeout,
+	})
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	fmt.Println("CONFIG:", configStr)
```
Comment (BenFradet) [inline sql_runner/snowflake_target.go]: maybe we should remove this?

**Comment by alexanderdean** [comment]
Thanks @BenFradet - if you could action Ben's feed @mbobrovsky that would be great.

File: sql_runner/snowflake_target.go (hunk 3 above)
Comment (mbobrovsky) [reply sql_runner/snowflake_target.go]: Yes, i did this, it's unnecessary here

File: sql_runner/snowflake_target.go (hunk 1 above)
Comment (mbobrovsky) [reply sql_runner/snowflake_target.go]: It's snowflake specific - it performs only one statement per request, so we should loop through all the statements. What i've done - i added a circuit on first error.

File: sql_runner/snowflake_target.go (hunk 2 above)
Comment (mbobrovsky) [reply sql_runner/snowflake_target.go]: No, I don't think so, int64 is just snowflakes' driver type of affected rows, I think we may do a type assertion to int to not change the other codebase.

File: sql_runner/snowflake_target.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+			res, err = sft.Client.Exec(script)
+
+			if err != nil {
+				return QueryStatus{query, query.Path, int(affected), err}
+			} else {
+				aff, _ := res.RowsAffected()
+				affected += aff
+			}
+		}
```
Comment (BenFradet) [inline sql_runner/snowflake_target.go]: should we return an error if `len(strings.TrimSpace(script)) <= 0`?

Comment (artemarta) [reply sql_runner/snowflake_target.go]: Why do you think we need it?

Comment (BenFradet) [reply sql_runner/snowflake_target.go]: it's always good to signal to the user that her/his input is malformed instead of silently dismissing it, no?

Comment (mbobrovsky) [reply sql_runner/snowflake_target.go]: Well, we can notify about the empty query string, or empty query file, but should we do it inside RunQuery() method?

Comment (BenFradet) [reply sql_runner/snowflake_target.go]: On second thought, this is just a matter of the last split being empty isn't it? In this case I think it's fine.

**Comment by alexanderdean** [comment]
@BenFradet am I right in thinking there are still changes for @mbobrovsky to make before you give the green tick?

**Comment by BenFradet** [comment]
@alexanderdean nope, just changed to approved

**Comment by alexanderdean** [comment]
Thanks scheduling.

**Comment by alexanderdean** [comment]
Hey @mbobrovsky - I started to intergrate this for UA testing before I noticed that Travis is failing:

https://travis-ci.org/snowplow/sql-runner/builds/318704666?utm_source=github_status&utm_medium=notification

The integration test suite is still using Golang 1.6 and 1.7, whereas it looks like the gosnowflake driver is only tested with 1.8 and 1.9:

https://github.com/snowflakedb/gosnowflake/blob/master/.travis.yml

You will need to update your PR so that tests pass (obviously). I've created a couple of tickets for you:

* https://github.com/snowplow/sql-runner/issues/112
* https://github.com/snowplow/sql-runner/issues/110

To ensure a clean git history, you must add these into your branch *before* the current Snowflake command.

Does that make sense? Let me know if any questions.

**Comment by mbobrovsky** [comment]
@alexanderdean What exactly should I do? As I undestood you commited changes on #110 , but not on #112 . Also, I can't figure out what branch is it on. 
So should I just update my project to this commit with updated golang version and than add all the commits I've made before for snowflake suport?
Please, let me know what branch I should update from.

**Comment by alexanderdean** [comment]
Hey @mbobrovsky:

* Don't worry about my release branch, continue to work in your PR
* I want to see three commits in your PR's branch, one each covering #110, #112 and #108 
* Travis needs to pass

Thanks!

**Comment by alexanderdean** [comment]
I believe `godep update -goversion` is what you need as part of #112

**Comment by BenFradet** [comment]
@mbobrovsky travis passes 👍 , could you rebase so that only 3 commits (corresponding to the 3 issues) are left:

- `Remove Go versions 1.6 and 1.7 from .travis.yml (closes #110)`
- `Bump Go version to 1.9.2 (closes #112)`
- `Add support for Snowflake database (closes #114)`

**Comment by mbobrovsky** [comment]
@BenFradet Did I do right?

**Comment by alexanderdean** [comment]
Rebase to only 3 commits please

**Comment by mbobrovsky** [comment]
Let's I create a new branch, and then create a new PR, it will be easier

**Comment by alexanderdean** [comment]
Go for it, closing this one

**Pull request closed without merging**